src/wad/image.rs

```diff
@@ -77,36 +87,51 @@ impl Image {
 
                 // The second byte is the length of this run. Skip an additional
                 // byte which is ignored for some reason.
-                let run_length =
-                    *try!(source.next()
-                                .to_err_with(|| format!("missing run length: column {}, run {}",
-                                                        i_column, i_run))) as usize;
+                let run_length = *try!(source.next()
+                                             .to_err_with(|| {
+                                                 format!("missing run length: column {}, run {}",
+                                                         i_column,
+                                                         i_run)
+                                             })) as usize;
 
                 // Check that the run fits in the image.
                 if row_start + run_length > height {
-                    return Err(format!("run out of bounds: column {}, run {} ({}, +{}), \
-                                        size {}x{}", i_column, i_run, row_start, run_length,
-                                       width, height).into());
+                    return Err(format!("run out of bounds: column {}, run {} ({}, +{}), size \
+                                        {}x{}",
+                                       i_column,
+                                       i_run,
+                                       row_start,
+                                       run_length,
+                                       width,
+                                       height)
+                                   .into());
```

Comment:
Do you have any advice on how to deal with this without adding so many lines? It's just the error handling branch, so, it's really breaking the flow here. I can't really move this to a separate function since it needs local variables :/ . If nothing comes to mind, I'm also okay with leaving it like this until I figure something out.